### ISSUE TITLE
Migrate TypeQLSteps implementation from client-python

### ISSUE THREAD

**Issue by alexjpwalker** [typedb/typedb-driver#317]
## Problem to Solve

TypeQLSteps is bulky, messy, hard to follow and hard to debug when things go wrong.

## Proposed Solution

I wrote an improved implementation, in particular of the `uniquely identify answer concepts` and `answer groups are` steps, in client-python. It is structured more cleanly, introducing the `ConceptMatcher` and `ConceptMatchResult` classes - the latter of which is really powerful for debugging failing tests. We should migrate this code to client-java (and client-nodejs).

**Comment by farost** [comment]
We made sure to minimize the BDD steps in 3.x, and we can't reuse these anymore. Sorry, Alex! Closing this
